lib/task.js

```diff
@@ -97,15 +98,20 @@ function factory(
         self.successStates = [TaskStates.Succeeded];
         // hint to whatever is running the task when it has failed
         self.failedStates = [TaskStates.Failed, TaskStates.Timeout, TaskStates.Cancelled];
-
+   
+        var server;
+        if (_.has(context, 'proxy')) {
+            server = context.proxy;
+        } else {
+            server = 'http://%s:%s'.format(
+                Task.configCache.apiServerAddress,
+                Task.configCache.apiServerPort
```

Comment:
I just realized this is not a new problem that brought by this PR, so this comment is not gating for me. If @WangWinson or @cgx027 confirms this is a problem need be resolved, I can accept move this work to another PR.